hack/generated/controllers/generic_controller.go

```diff
@@ -364,27 +369,29 @@ func (gr *GenericReconciler) MonitorDelete(
 }
 
 func (gr *GenericReconciler) CreateDeployment(ctx context.Context, action ReconcileAction, data *ReconcileMetadata) (ctrl.Result, error) {
-	deployment, err := gr.resourceSpecToDeployment(ctx, data)
+	toDeploy, err := gr.resourceSpecToDeployment(ctx, data)
 	if err != nil {
 		return ctrl.Result{}, err
 	}
 
-	// TODO: Could somehow have a method that grouped both of these calls
-	data.log.Info("Starting new deployment to Azure", "action", string(action), "id", deployment.Id)
-	gr.Recorder.Event(data.metaObj, v1.EventTypeNormal, string(action), fmt.Sprintf("Starting new deployment to Azure with ID %q", deployment.Id))
-
-	err = gr.Patch(ctx, data, func(ctx context.Context, mutData *ReconcileMetadata) error {
-		deployment, err = gr.ARMClient.CreateDeployment(ctx, deployment)
-		if err != nil {
-			return err
-		}
+	// Try to create deployment:
+	data.log.Info("Starting new deployment to Azure", "action", string(action))
+	deployment, err := gr.ARMClient.CreateDeployment(ctx, toDeploy)
 
-		err = mutData.Update(deployment, nil) // Status is always nil here
-		if err != nil {
-			return err
+	if err != nil {
+		var reqErr *autorestAzure.RequestError
+		if errors.As(err, &reqErr) && reqErr.StatusCode == 409 { /* == Conflict */
```

Comment:
Could use this: https://golang.org/src/net/http/status.go

That way it reads better.

Reply:
Also is there a TODO here where we need to diff the deployment we want to create and the existing deployment and do... something?

As a matter of fact it might be worth leaving a bit of an explanation here as to how a collision can even happen and what our "desired" behavior in that case is.

Reply:
Yes, I noted that in the PR description, but I'll make it explicit in the code.